lockfile_generator/src/gradle.rs

```diff
@@ -24,4 +25,14 @@ impl Generator for Gradle {
     fn tool(&self) -> &'static str {
         "Gradle"
     }
+
+    fn check_prerequisites(&self, manifest_path: &Path) -> Result<()> {
+        if manifest_path.file_name() != Some(OsStr::new("build.gradle"))
+            && manifest_path.file_name() != Some(OsStr::new("build.gradle.kts"))
```

Comment:
It took my brain a minute to confirm that this condition is correctly written with `&&` instead of `||`. Negatives are hard...

This might be easier to read if the condition was reversed (i.e., `if <correct file name> { Ok(()) } else { Err(...) }`)

Or it could be written with a `match` statement like so:

```patch
diff --git a/lockfile_generator/src/gradle.rs b/lockfile_generator/src/gradle.rs
index ea3f7d7..3b0b7f7 100644
--- a/lockfile_generator/src/gradle.rs
+++ b/lockfile_generator/src/gradle.rs
@@ -1,6 +1,5 @@
 //! Java gradle ecosystem.

-use std::ffi::OsStr;
 use std::path::{Path, PathBuf};
 use std::process::Command;

@@ -27,12 +26,9 @@ impl Generator for Gradle {
     }

     fn check_prerequisites(&self, manifest_path: &Path) -> Result<()> {
-        if manifest_path.file_name() != Some(OsStr::new("build.gradle"))
-            && manifest_path.file_name() != Some(OsStr::new("build.gradle.kts"))
-        {
-            Err(Error::InvalidManifest(manifest_path.to_path_buf()))
-        } else {
-            Ok(())
+        match manifest_path.file_name().and_then(|f| f.to_str()) {
+            Some("build.gradle" | "build.gradle.kts") => Ok(()),
+            _ => Err(Error::InvalidManifest(manifest_path.to_path_buf())),
         }
     }
 }
```

But it's not terrible as is, so I won't hold up this review.

Reply:
Funny...I thought the same thing and had it the inverted way initially but then thought maybe _that_ would be a problem later if someone wants to add new prerequisites and the pattern of "failure cases first" was not in place. It has been changed.